jetty-server/src/main/java/org/eclipse/jetty/server/HttpChannel.java

```diff
@@ -548,27 +550,28 @@ protected void handleException(Throwable x)
             {
                 // Handle error normally
                 _request.setHandled(true);
-                _request.setAttribute(RequestDispatcher.ERROR_EXCEPTION, x);
-                _request.setAttribute(RequestDispatcher.ERROR_EXCEPTION_TYPE, x.getClass());
+                _request.setAttribute(RequestDispatcher.ERROR_EXCEPTION, failure);
+                _request.setAttribute(RequestDispatcher.ERROR_EXCEPTION_TYPE, failure.getClass());
 
                 if (isCommitted())
                 {
-                    abort(x);
+                    abort(failure);
                     if (LOG.isDebugEnabled())
-                        LOG.debug("Could not send response error 500, already committed", x);
+                        LOG.debug("Could not send response error 500, already committed", failure);
                 }
                 else
                 {
                     _response.setHeader(HttpHeader.CONNECTION.asString(), HttpHeaderValue.CLOSE.asString());
 
-                    if (x instanceof BadMessageException)
+                    Throwable cause = unwrap(failure);
+                    if (cause instanceof BadMessageException)
                     {
-                        BadMessageException bme = (BadMessageException)x;
+                        BadMessageException bme = (BadMessageException)cause;
                         _response.sendError(bme.getCode(), bme.getReason());
                     }
-                    else if (x instanceof UnavailableException)
+                    else if (cause instanceof UnavailableException)
```

Comment:
This will never happen because `UnavailableException` is-a `ServletException`.
So if you really need to unwrap deeply, remove this branch. Otherwise if you need to handle `UnavailableException`, then the unwrapping should be modified.